File: frontend/src/types/session.ts
export interface Session {
  id: number;
  user_id: number;
  exercise: string;
  reps: number;
  form_accuracy: number;
  duration_sec: number;
  notes: string;
  date: string | null;
}

export interface SessionSummary {
  exercise: string;
  total_sessions: number;
  total_reps: number;
  avg_form: number;
}

export interface ProgressData {
  total_sessions: number;
  total_reps: number;
  avg_form_accuracy: number;
  exercises_done: number;
  summary: SessionSummary[];
  recent_sessions: Session[];
}

export interface ProgressSummary {
  total_sessions: number;
  total_reps: number;
  avg_form_accuracy: number;
  exercises_done: number;
  current_streak: number;
}

File: backend/schemas/session.py
"""
Pydantic schemas for session data.
"""

from pydantic import BaseModel
from typing import Optional


class SessionResponse(BaseModel):
    id: int
    user_id: int
    exercise: str
    reps: int
    form_accuracy: int
    duration_sec: int = 0
    notes: str = ""
    date: Optional[str] = None


class SessionSummary(BaseModel):
    exercise: str
    total_sessions: int
    total_reps: int
    avg_form: float


class ProgressResponse(BaseModel):
    total_sessions: int
    total_reps: int
    avg_form_accuracy: float
    exercises_done: int
    summary: list[SessionSummary]
    recent_sessions: list[SessionResponse]
